dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -1642,16 +1650,27 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
         }
         for (String missing : result.getMissing()) {
             reportedErrors = true;
+            boolean isRootFeature = rootFeatures.contains(missing);
+            boolean isExtension = missing.indexOf(":") > -1;
+            String altName = featureRepository.matchesAlternate(missing);
             //Check if using Open Liberty before suggesting install util for missing features
             if (!getProductInfoDisplayName().startsWith(PRODUCT_INFO_STRING_OPEN_LIBERTY)) {
-                if (rootFeatures.contains(missing) && missing.indexOf(":") < 0) {
+                if (isRootFeature && !isExtension) {
```

Comment:
Why is this check only done when not on Open Liberty?

The check for `!isExtension` should be done done regardless if we are on Open Liberty or not.  I suggest the following check be done after this if-then-else block entirely so it applies to both:
```
if (altName != null && isRootFeature && !isExtension) {
    Tr.error(tc, "MISSING_FEATURE_HAS_ALT_NAME", missing, altName);
}
```

Reply:
It's unlikely that an extension feature (ie a feature containing an ':' in the name) will match a key in the atlNames map so the check is redundant for that purpose. The isExtension check in the code was to select between the two flavors of the UPDATE_MISSING_FEATURE_ERROR_<*> message. Open Liberty does not have both error messages. 

I'll restructure to print the ALT_NAME message outside the scope of the is-closed-liberty conditional block:
```
            //Check if using Open Liberty before suggesting install util for missing features
            if (!getProductInfoDisplayName().startsWith(PRODUCT_INFO_STRING_OPEN_LIBERTY)) {
                if (isRootFeature && !isExtension) {
                    // Only report this message for core features included as root features in the server.xml
                    Tr.error(tc, "UPDATE_MISSING_CORE_FEATURE_ERROR", missing, locationService.getServerName());
                } else {
                    Tr.error(tc, "UPDATE_MISSING_FEATURE_ERROR", missing);
                }
            } else {
                // Not on Open Liberty
                Tr.error(tc, "UPDATE_MISSING_FEATURE_ERROR", missing);
            }
            if (altName != null && isRootFeature) {
                Tr.error(tc, "MISSING_FEATURE_HAS_ALT_NAME", missing, altName);
            }
            installStatus.addMissingFeature(missing);
```